stl/inc/system_error

```diff
@@ -554,9 +553,31 @@ _NODISCARD const _Ty& _Immortalize_memcpy_image() noexcept {
 #elif defined(__cpp_constexpr_dynamic_alloc)
 template <class _Ty>
 _NODISCARD const _Ty& _Immortalize_memcpy_image() noexcept {
-    static constexpr _Ty _Static;
+    static _CONSTEVAL _Ty _Static;

```

Comment:
WG21-N4861 9.2.5 [[dcl.constexpr]/1](https://eel.is/c++draft/dcl.constexpr#1): "The `consteval` specifier shall be applied only to the declaration of a function or function template."

Did you mean "`constinit` falling back to `constexpr`" here? Do we need `constinit` at all (I thought `constexpr` was stronger)?

In general, I am very nervous about "automatically lighting-up code" that currently doesn't activate for any compilers - how do we know that it's valid, versus causing an error the first time that a user compiles an older STL with a newer compiler?

Reply:
> In general, I am very nervous about "automatically lighting-up code" that currently doesn't activate for any compilers

Clang uses this code path; Clang 10 defines `__cpp_constexpr_dynamic_alloc`. Clang 9 did not, which allowed the "bad" code to live here before.

Reply:
Yep, that's my concern exactly - the code was ultimately inactive.

Reply:
@CaseyCarter Can you push what this should be into the branch? Otherwise I'd prefer to put the `[[clang::require_constant_initialization]]` version I had before.

Reply:
The `constexpr` specifier requires constant-initialization when applied to the definition of a variable with static or thread storage duration. I don't think there's anything more to do here.